src/net/sourceforge/kolmafia/request/AdventureRequest.java

```diff
@@ -1065,6 +1065,12 @@ public static final void handleServerRedirect(String redirectLocation) {
       return;
     }
 
+    if (redirectLocation.startsWith("shop.php")) {
+      // The Shore Inc. can redirect to the gift shop.
+      AdventureRequest.ZONE_UNLOCK.run();
+      return;
+    }
+
```

Comment:
I wish we could handle this (and the three identical code blocks above it) with something like a switch.  Are we far enough along on the Java version to get a better switch statement?

Reply:
> Is there a unit test that could be written for this function?

I'm not writing a unit test.

Reply:
I have tried, there's no good test here for now.